---
int32 res
string error
